iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/DanglingLineBoundaryImpl.java

```diff
@@ -18,70 +18,74 @@ public class DanglingLineBoundaryImpl implements Boundary {
+    private final DanglingLine parent;
 
     public DanglingLineBoundaryImpl(DanglingLine parent) {
         this.parent = Objects.requireNonNull(parent);
     }
 
-    public void setSide(Branch.Side side) {
-        this.side = side;
-    }
-
     @Override
     public double getV() {
-        if (valid(parent.getP0(), parent.getQ0())) {
+        if (!parent.isMerged() && valid(parent.getP0(), parent.getQ0())) {
             DanglingLineData danglingLineData = new DanglingLineData(parent, true);
             return danglingLineData.getBoundaryBusU();
         }
 
         Terminal t = parent.getTerminal();
         Bus b = t.getBusView().getBus();
-        return new SV(t.getP(), t.getQ(), getV(b), getAngle(b), Optional.ofNullable(side).orElse(Branch.Side.ONE)).otherSideU(parent, true);
+        return new SV(t.getP(), t.getQ(), getV(b), getAngle(b), Branch.Side.ONE).otherSideU(parent, true);
     }
 
     @Override
     public double getAngle() {
-        if (valid(parent.getP0(), parent.getQ0())) {
+        if (!parent.isMerged() && valid(parent.getP0(), parent.getQ0())) {
             DanglingLineData danglingLineData = new DanglingLineData(parent, true);
             return Math.toDegrees(danglingLineData.getBoundaryBusTheta());
         }
         Terminal t = parent.getTerminal();
         Bus b = t.getBusView().getBus();
-        return new SV(t.getP(), t.getQ(), getV(b), getAngle(b), Optional.ofNullable(side).orElse(Branch.Side.ONE)).otherSideA(parent, true);
+        return new SV(t.getP(), t.getQ(), getV(b), getAngle(b), Branch.Side.ONE).otherSideA(parent, true);
     }
 
     @Override
     public double getP() {
-        if (valid(parent.getP0(), parent.getQ0())) {
+        if (!parent.isMerged() && valid(parent.getP0(), parent.getQ0())) {
             return -parent.getP0();
         }
         Terminal t = parent.getTerminal();
         Bus b = t.getBusView().getBus();
-        return new SV(t.getP(), t.getQ(), getV(b), getAngle(b), Optional.ofNullable(side).orElse(Branch.Side.ONE)).otherSideP(parent, true);
+        return new SV(t.getP(), t.getQ(), getV(b), getAngle(b), Branch.Side.ONE).otherSideP(parent, true);
     }
 
     @Override
     public double getQ() {
-        if (valid(parent.getP0(), parent.getQ0())) {
+        if (!parent.isMerged() && valid(parent.getP0(), parent.getQ0())) {
             return -parent.getQ0();
         }
         Terminal t = parent.getTerminal();
         Bus b = t.getBusView().getBus();
-        return new SV(t.getP(), t.getQ(), getV(b), getAngle(b), Optional.ofNullable(side).orElse(Branch.Side.ONE)).otherSideQ(parent, true);
+        return new SV(t.getP(), t.getQ(), getV(b), getAngle(b), Branch.Side.ONE).otherSideQ(parent, true);
     }
 
-    @Override
+    /**
+     * @deprecated Not used anymore.
+     */
+    @Deprecated(since = "5.2.0")
     public Branch.Side getSide() {
-        return side;
+        throw new PowsyblException("Deprecated. Not used anymore");
     }
 
```

Comment:
You can delete this method (for this PR, since the breaking change is quite important, we don't ensure backwards compatibility)

Reply:
Done